java/src/jmri/InstanceManager.java

```diff
@@ -773,6 +860,38 @@ public InstanceManager() {
         }
     }
 
+
+    /**
+     * Get a list of all registered objects of a specific type.
+     *
+     * Intended for use with i.e. scripts where access to the class type is inconvenient.
+     *
+     * @param <T>  type of the class
+     * @param className Fully qualified class name
+     * @return a list of registered instances with the manager or an empty
+     *         list
+     */
+    @SuppressWarnings("unchecked") // the cast here is protected by the structure of the managerLists
+    @Nonnull
+    public <T> List<T> getInstances(@Nonnull String className) {
+        Class<?> type;
+        try {
+            type = Class.forName(className);
+        } catch (ClassNotFoundException ex) {
+            log.error("No class found: {}", className);
+            throw new IllegalArgumentException(ex);
```

Comment:
I think it would be good to add `@throws IllegalArgumentException` to the Javadoc for this method.
One possible interpretation of `getInstances()` is that it would return an empty list if the class name is wrong. By adding `@throws` to the Javadoc, it's clear what happens then a bad class name is given.